modules/gapi/src/api/gmeta_check.cpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include "precomp.hpp"
+
+#include <opencv2/gapi/util/throw.hpp>
+#include "api/gmeta_check.hpp"
+
+namespace cv
+{
+void validate_input_meta_arg(const GMetaArg& meta)
+{
+    switch (meta.index())
+    {
+        case GMetaArg::index_of<GMatDesc>():
+        {
+            validate_input_meta(util::get<GMatDesc>(meta));
+        }
```

Comment:
this `case` requires `break` too; it doesn't break now but may break in the future (as somebody will copy-paste this part blindly to introduce a new check for some another type).

Reply:
agree, it's strange that no CI build scenario failed with compilation warning